### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+language: node_js
+node_js:
+  - '0.12'
+sudo: false
+env:
+  matrix:
+    - GROUP=
+  global:
+    - secure: CVBwXgZPtMI4nEK0MmU1x9weawNvHtYeK4TjXx9FyrVfIOW1eOvafLirUK+efZllme6nRElWFcFmJzHdkVKmeLR/IpT+EfS6LNsDVyHeLvCKIVyug7hTBZXsBCp1cEcevAMJnC6miM8VFUFswZv1LAiThF8avNNkFbjejygsoCMDl3HLnp00sPSdCN664kTDv1EOIF4YH/aR812Ow9L7Nxy/goiuQWp+rQVTddrVaqsTC94kNK3Q1YZ/EAIEOvSR6eS05fTkhAbciJmguvxJpgIyOZ8IPkyRNPZ0Pq9XwdkyLY2rEKqlIdd4optwwi3IUumIQAZ0Ox1I/+Z+jKiPCHOxn1i4eodwyG9ic0BqoZI0IkRP7C7ndgoPUO/o2QSoFqOq3uRPwge1jv1QU0eR7BsG/EIlD8yWpFyufmf1hVQgyHybS3A7LnBBHCQwOiPchMk9fBR0EEMMX9qWImZIxZZYMg3URmQe/ze2302H4SPqq+Q7hv2sw8c6hNpE2eJpQtHWc94PPbq4adbi1JoYKTdwL8gLRPH3QhXHswB1RbHrY2OnZwHU7K3Z3aDjV3/bbAmp7geyYHK55U5l6FRjrjl241lHEicUIjJWPa5c9wFzVtf2Tzpo7AQzeGNUez2G7ofZOjEpVFDWMFsxzjXCDbtAnsKfgtg2RUt7kbXGiWo=
```
Comment (sccolbert) [inline .travis.yml]: remove secure token (I'll make a new one).

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+*.suo
+*.user
+.DS_Store
+docs/api
+lib
+node_modules
+test/build
+test/coverage
+example/phosphor-plugins
```
Comment (sccolbert) [inline .gitignore]: what's the purpose of ignoring example/phosphor-plugins?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,94 @@
-# phosphor-plugins
+phosphor-plugins
+==================
```
Comment (sccolbert) [inline README.md]: header lines same length as header text

File: README.md [hunk 4]
```diff
@@ -1,2 +1,94 @@
...
 A module for building plugin-based applications.
+
+
+[![Build Status](https://travis-ci.org/phosphorjs/phosphor-plugins.svg)](https://travis-ci.org/phosphorjs/phosphor-plugins?branch=master)
+[![Coverage Status](https://coveralls.io/repos/phosphorjs/phosphor-plugins/badge.svg?branch=master&service=github)](https://coveralls.io/github/phosphorjs/phosphor-plugins?branch=master)
+
+Slightly opinionated scaffolding for building plugin-based IDE-style applications..
```
Comment (sccolbert) [inline README.md]: Replace this line with the description text from the original readme.

File: example/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    lib.loadPlugin('foo').then(() => {
+        console.log('foo finished loading');
+        lib.loadPlugin('bar').then(() => {
+            console.log('bar finished loading');
+            lib.unloadPlugin('bar');
+            lib.unloadPlugin('foo');
+            console.log('all plugins unloaded');
+        });
+    });
+});
```
Comment (sccolbert) [inline example/index.js]: chain promises by returning the previous promise, rather than using nested closures.

File: example/package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "phosphide-demo",
```
Comment (sccolbert) [inline example/package.json]: wrong package name

File: package.json [hunk 7]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "phosphor-plugins",
+  "version": "0.1.0",
+  "description": "A slightly opinionated scaffolding for building plugin-based IDE-style applications.",
```
Comment (sccolbert) [inline package.json]: use the description from the original readme (which is also the repo description).

File: package.json [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  "main": "lib/index.js",
+  "typings": "lib/index.d.ts",
+  "dependencies": {
+    "phosphor-disposable": "^1.0.4"
+  },
+  "devDependencies": {
+    "coveralls": "^2.11.4",
+    "expect.js": "^0.3.1",
+    "glob-copy": "^0.1.0",
```
Comment (sccolbert) [inline package.json]: is glob copy an actual dependency?

File: package.json [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/phophorjs/phosphor-plugins.git"
+  },
+  "files": [
+    "lib/*.js",
+    "lib/*.d.ts"
+  ],
+  "keywords": [
+    "phosphide",
```
Comment (sccolbert) [inline package.json]: no phosphide keywork

File: package.json [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    "plugin",
+    "application"
+  ],
+  "author": "Dave Willmer <dave.willmer@gmail.com>",
```
Comment (sccolbert) [inline package.json]: Correct author?

File: scripts/travis_after_success.sh [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+# push docs on master branch
+if [[ $TRAVIS_PULL_REQUEST == false && $TRAVIS_BRANCH == "master" ]]
+then
+    echo "-- will push docs --"
+
+    git config --global user.email "travis@travis-ci.com"
+    git config --global user.name "Travis Bot"
+
+    git clone https://github.com/phosphorjs/phosphor-widget.git travis_docs_build
```
Comment (sccolbert) [inline scripts/travis_after_success.sh]: wrong url

File: .gitignore (hunk 2 above)
Comment (blink1073) [inline .gitignore]: We use glob-copy to move the `lib/` files to this folder for the example.

**Comment by blink1073** [comment]
All comments addressed (but not pushed), up to this point.

File: .gitignore (hunk 2 above)
Comment (sccolbert) [inline .gitignore]: are you not able to relative import them like the other Phosphor examples? or do you want the example to be "as-installed"?

This, by the way, is one of my few complaints with NPM/Node, in that you can import yourself from within your own package.

**Comment by sccolbert** [comment]
Looks good enough to start iterating. We'll need to add validation for the plugin JSON, ensure no duplicate extension points, etc...

**Comment by sccolbert** [comment]
Push your changes, I'll merge and start opening issues and PRs.

**Comment by blink1073** [comment]
I pushed the changes.  I needed to copy the files because you need to run the server from `example` in order for steal.js to pick up `node_modules`, but the server won't serve files from a parent directory.

**Comment by sccolbert** [comment]
Got it. Thanks!

**Pull request merged**